src/Transform/ONNX/ONNXOpTransformPass.cpp

```diff
@@ -135,6 +135,7 @@ void ONNXOpTransformPass::runOnOperation() {
     dynamicPM.addNestedPass<FuncOp>(onnx_mlir::createDecomposeONNXToONNXPass());
     dynamicPM.addPass(onnx_mlir::createShapeInferencePass());
     dynamicPM.addPass(mlir::createCanonicalizerPass());
+    dynamicPM.addPass(onnx_mlir::createShapeInferencePass());
```

Comment:
Why need to add shape inference pass here?

Reply:
Because canonicalization can (and in this case does) introduce operators that might have unknown shape for their output. In other places in the code where canonicalization is called we also do shape inference before calling it.